### DISCUSSION TITLE
Mixed Content https error when redirecting and using custom response.headers

### DISCUSSION THREAD

**Discussion by diegocgaona** [fastapi/fastapi#8706] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# my middleware options
app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
    expose_headers=["X-Total-Count"],
)

app.add_middleware(GZipMiddleware, minimum_size=1000)
# Tried the bellow, but get error because CORS preflight for options can't have redirects
# app.add_middleware(HTTPSRedirectMiddleware)


# My endpoint example
@router.get("/", response_model=responseschema)
    async def read_job(item_id: int,
                       user: UserDB = Depends(current_active_user)):
    response.headers['X-Total-Count'] = str(await model.all().count())
    return await my_response
```

### Description

When I call my endpoint from my front (both using https:// and the front is in the allowed origins)  without "/", it does the redirect and then I receive the error:

`Mixed Content: The page at 'https://front.mydomain.com/#/clients/1' was loaded over HTTPS, but requested an insecure resource 'http://back.mydomain/jobs/?_end=25&_order=DESC&_sort=id&_start=0&client_id=1'. This request has been blocked; the content must be served over HTTPS.`

I think it comes from my exposed Header (`response.headers['X-Total-Count'] = str(await model.all().count())`), 
![image](https://user-images.githubusercontent.com/9142126/146218183-3dc011d1-1bfa-4347-bbab-a6a83afb0f12.png)

It works fine if I don't have the 307 redirect, which occurs when my front uses: `https://back.mydomain/jobs?` instead of `https://back.mydomain/jobs/?`. The the header location response as `http://back.mydomain/jobs/?` without https://.

How can I make the response header location come with https:// (or not come, because in the OK answer without redirect I don't see this location) after the redirect?

Bellow a print of a correct call (without the redirect and do not have location):

![image](https://user-images.githubusercontent.com/9142126/146219921-06e24219-f373-4bf1-8574-3ddd5f53d0a7.png)

Someone had a similar problem? I solved some problems changing the front, but in this case I can't.

### Operating System

Linux

### Operating System Details

Is an Linux (Ubuntu 20.04) GCE instance on Google Cloud.

### FastAPI Version

0.70.0

### Python Version

3.9

### Additional Context

Everything is fine, but only have problem with this redirects.
I'm using an Docker with Traefik (mostly because of SSL).

**Comment by aalokpatwa-athelas** [comment]
Hey @diegocgaona, did you ever find an answer to this question? Having the same issue.

**Comment by diegocgaona** [comment]
@aalokpatwa-athelas If I remember well, what I did was change my front and my endpoints to not use the "/" in the end.
It's not a real/ideal solution... but was the only way I could make it work.

**Comment by zwessels** [comment, marked as answer]
for anybody else that comes here, https://stackoverflow.com/questions/63511413/fastapi-redirection-for-trailing-slash-returns-non-ssl-link
